fundamentals/android/auth-sign-in-with-google/Notes/app/src/main/java/com/notes/app/NotesAppState.kt

```diff
@@ -2,9 +2,27 @@ package com.notes.app
 
 import androidx.compose.runtime.Stable
 import androidx.navigation.NavHostController
+import androidx.compose.material3.SnackbarHostState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
 
 @Stable
-class NotesAppState(val navController: NavHostController) {
+class NotesAppState(
+    private val snackbarHostState: SnackbarHostState,
+    val navController: NavHostController,
+    private val snackbarManager: SnackbarManager,
+    coroutineScope: CoroutineScope
+) {
+    init {
+        coroutineScope.launch {
+            snackbarManager.snackbarMessages.filterNotNull().collect { message ->
+                snackbarHostState.showSnackbar(message)
+                snackbarManager.clearSnackbarState()
+            }
+        }
+    }
```

Comment:
Not blocking: this is a scary bit of code, it seems like the collection from this flow starts when the app starts? And stops when... Well, it's not clear because it's not tied to any lifecycle, but probably when the app stops?
Let's review this in the future.

Reply:
Yes, it's supposed to run until the app stops, as it is used to display snackbar to the user in any of the screens. It was reviewed by the Android team on the Make it So app (I'm using the same SnackbarManager) - but we can definitely review this in the future as there probably is a better way to launch Snackbars these days!